orderer/common/cluster/comm.go

```diff
@@ -658,20 +766,30 @@ func requestAsString(request *orderer.StepRequest) string {
 
 // NewStream creates a new stream.
 // It is not thread safe, and Send() or Recv() block only until the timeout expires.
-func (rc *RemoteContext) NewStream(timeout time.Duration) (*Stream, error) {
+func (rc *RemoteContext) NewStream(timeout time.Duration, Signer identity.SignerSerializer) (*Stream, error) {
 	if err := rc.ProbeConn(rc.conn); err != nil {
 		return nil, err
 	}
 
 	ctx, cancel := context.WithCancel(context.TODO())
-	stream, err := rc.Client.Step(ctx)
+
+	var stream grpc.ClientStream
+	var err error
+	switch cli := rc.Client.(type) {
+	case orderer.ClusterClient:
+		stream, err = cli.Step(ctx)
+	case orderer.ClusterNodeServiceClient:
+		stream, err = cli.Step(ctx)
+	default:
+		cancel()
+		return nil, errors.New("cluster client type not valid")
+	}
 	if err != nil {
 		cancel()
 		return nil, errors.WithStack(err)
 	}
-
-	streamID := atomic.AddUint64(&rc.nextStreamID, 1)
 	nodeName := commonNameFromContext(stream.Context())
+	streamID := atomic.AddUint64(&rc.nextStreamID, 1)
```

Comment:
why did you change places with nodeName?

Reply:
sorry, it was not intentional, i will revert it back.